### PR TITLE
[csharp-netcore] Update RestSharp and improve async methods

### LINKED ISSUE #3066: [BUG] [csharp-netcore] Obsolete method warning
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X] What's the version of OpenAPI Generator used?
- [X] Have you search for related issues/PRs?
- [X] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

The `RestSharp` `RestClient` throws a warning about deprecated/obsolete 
`csharp-netcore` `tfm` `2.0`
```
1>Client\ApiClient.cs(385,17,385,133): warning CS0618: 'RestClient.AddHandler
(IDeserializer, params string[])' is obsolete: 
'Use the overload that accepts a factory delegate'
1>Client\ApiClient.cs(390,17,390,118): warning CS0618: 'RestClient.AddHandler(
IDeserializer, params string[])' is obsolete: 
'Use the overload that accepts a factory delegate'
1>Client\ApiClient.cs(393,13,393,98): warning CS0618: 'RestClient.AddHandler(
IDeserializer, params string[])' is obsolete: 
'Use the overload that accepts a factory delegate'
```
##### openapi-generator version

`master`

##### OpenAPI declaration file content or url
no special yaml needed. Even default petstore cause same issue.

### PR DESCRIPTION
**Pull request by shibayan**
Update RestSharp and streamline execution of task-based asynchronous methods. Removed deprecated blocking code for synchronous methods.

This PR close #3066

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by etherealjoy** [comment]
@shibayan 
Thanks for fixing the issue, I assume this warning is also removed right? It was not originally listed in #3066
```
ApiClient.cs(412,34,412,65): warning CS0618: 'RestClient.ExecuteTaskAsync<T>(IRestRequest)' is obsolete: 'Please use ExecuteAsync instead'
```

**Comment by shibayan** [comment]
@etherealjoy 
Yes. At the same time, `ExecuteTaskAsync<T>` => `ExecuteAsync<T>` was replaced.

File: modules/openapi-generator/src/main/resources/csharp-netcore/ApiClient.mustache [hunk 1]
```diff
@@ -459,10 +517,10 @@ namespace {{packageName}}.Client
         /// <param name="configuration">A per-request configuration object. It is assumed that any merge with
         /// GlobalConfiguration has been done before calling this method.</param>
         /// <returns>A Task containing ApiResponse</returns>
-        public async Task<ApiResponse<T>> GetAsync<T>(string path, RequestOptions options, IReadableConfiguration configuration = null)
+        public Task<ApiResponse<T>> GetAsync<T>(string path, RequestOptions options, IReadableConfiguration configuration = null)
```
Comment (jimschubert) [inline modules/openapi-generator/src/main/resources/csharp-netcore/ApiClient.mustache]: Looks like this is a breaking change and will need to target 4.3.x

Comment (shibayan) [reply modules/openapi-generator/src/main/resources/csharp-netcore/ApiClient.mustache]: This is not a breaking change. C# async keyword has no effect on method signatures.

Comment (jimschubert) [reply modules/openapi-generator/src/main/resources/csharp-netcore/ApiClient.mustache]: Awesome. Thanks for clarifying this!

**Pull request merged**